src/TLuaInterpreter.cpp

```diff
@@ -11728,35 +11728,264 @@ int TLuaInterpreter::sendSocket( lua_State * L )
-    else
-    {
+    string who, text;
+    if( ! lua_isstring( L, 1 ) ) {
+        lua_pushfstring(L, "sendIrc: bad argument #1 type (Target as string expected, got %s!)", lua_typename(L, 1));
+        return lua_error(L);
+    } else {
         who = lua_tostring( L, 1 );
     }
-    string text;
-    if( ! lua_isstring( L, 2 ) )
-    {
-        lua_pushstring( L, "sendIrc: wrong argument type" );
-        lua_error( L );
-        return 1;
-    }
-    else
-    {
+    if( ! lua_isstring( L, 2 ) ) {
+        lua_pushfstring(L, "sendIrc: bad argument #2 type (Message as string expected, got %s!)", lua_typename(L, 2));
+        return lua_error(L);
+    } else {
         text = lua_tostring( L, 2 );
     }
-    QString chan = who.c_str();
-    QString txt = text.c_str();
-    if( ! mudlet::self()->mpIRC ) return 0;
-    mudlet::self()->mpIRC->connection->sendCommand( IrcCommand::createMessage( chan, txt ) );
-    return 0;
+    QString target = who.c_str();
+    QString msg = text.c_str();
+    Host* pHost = &getHostFromLua(L);
+    if ( !mudlet::self()->mpIrcClientMap[pHost] ) {
+        // create a new irc client if one isn't ready.
+        mudlet::self()->mpIrcClientMap[pHost] = new dlgIRC(pHost);
+        mudlet::self()->mpIrcClientMap[pHost]->raise();
+        mudlet::self()->mpIrcClientMap[pHost]->show();
+    }
+
+    // wait for our client to be ready before sending messages.
+    if (!mudlet::self()->mpIrcClientMap[pHost]->mReadyForSending) {
+        lua_pushboolean(L, false);
+        return 2;
+    }
+
+    lua_pushboolean(L, mudlet::self()->mpIrcClientMap[pHost]->sendMsg(target, msg));
+    return 1;
+}
+
+/** ircGetNick();
+ *  Returns a string containing the IRC client nickname.
+ */
+int TLuaInterpreter::ircGetNick(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QString nick;
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        nick = mudlet::self()->mpIrcClientMap[pHost]->getNickName();
+    } else {
+        nick = dlgIRC::readIrcNickName(pHost);
+    }
+
+    lua_pushstring(L, nick.toUtf8().constData());
+    return 1;
+}
+
+/** ircGetServer();
+ *  Returns a pair containing the IRC server and port.
+ */
+int TLuaInterpreter::ircGetServer(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QString hname;
+    int hport;
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        hname = mudlet::self()->mpIrcClientMap[pHost]->getHostName();
+        hport = mudlet::self()->mpIrcClientMap[pHost]->getHostPort();
+    } else {
+        hname = dlgIRC::readIrcHostName(pHost);
+        hport = dlgIRC::readIrcHostPort(pHost);
+    }
+
+    lua_pushstring(L, hname.toUtf8().constData());
+    lua_pushinteger(L, hport);
+    return 2;
+}
+
+/** ircGetChannels();
+ *  Returns a string containing channels separated by a space.
+ *  If a client is active the list contains channels currently joined.
+ *  Otherwise the list is read from IRC client settings.
+ */
+int TLuaInterpreter::ircGetChannels(lua_State* L)
```

Comment:
Could this return a proper Lua indexed table? If the person does really want it as a string they could easily do `table.concat(ircGetChannels(), ' ')`